### PR TITLE
Move encoding name to const string value

### PR DESCRIPTION
**Pull request by rstm-sf**
- [x]  part of multibyte
- [x]  part of singlebyte

### CODE REVIEW

**Comment by rstm-sf** [comment]
@304NotModified, hello!

> fix encoding name to supported

do you think this task should be made separate pr?

**Comment by 304NotModified** [comment]
I think that's a good idea :)

**Comment by rstm-sf** [comment]
> I think that's a good idea :)

Ok then everything is ready

File: src/Core/Charset.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+        internal const string UTF16_LE = "utf-16le";
+        
+        internal const string UTF16_BE = "utf-16be";
+        
+        internal const string UTF32_LE = "utf-32le";
+        
+        internal const string UTF32_BE = "utf-32be";
+
+        internal const string X_ISO_10646_UCS_4_3412 = "X-ISO-10646-UCS-4-3412"; // TODO: not supported?
```
Comment (304NotModified) [inline src/Core/Charset.cs]: please move these comments to XML comments (summary or remarks).

Then it's visible for other places and easy to refactor

Comment (rstm-sf) [reply src/Core/Charset.cs]: Ready ;)

**Comment by HelloWorld017** [comment]
Although it is totally irrelevant to this Pull-Request, as I saw this comment:
`TODO: Not supported? Maybe fix to ks-c-5601?`
I wanted to add some comments about why I used `CP949` instead of `ks_c_5601-1987`.

(These things are added in Pull Request #59)

----
The main reason I used `CP949` instead of `ks_c_5601-1987` is that `CP949` is the de-facto naming.

The name `ks_c_5601-1987` is barely recognized by another program. Even the famous program `iconv`, which is shipped by most linux distros, doesn't understand the charset `ks_c_5601-1987`. (They only recognize `CP949`) So the ffmpeg and many other programs that have dependency on `iconv` doen't recognize that too.

Actually, the name `ks_c_5601-1987` is recognized only by some Microsoft programs, for example: Internet Explorer, Outlook, ...., but most of them also recognize `CP949`.

----
Unfortunately, the `Encoding::GetEncoding` did not understand `CP949`. So to ensure this library work with other program and have backward-compatibility, I decided to make it return `CP949` as EncodingName and `Encoding::GetEncoding("ks_c_5601-1987")` as the Encoding object. And to make it work like that, I added alternative naming.

For example, when `CharsetProber::GetCharsetName` returns `name (alternative name)`, the Encoding object is got from the String `alternative name` and it returns `name` as the EncodingName.

This is why the `CP949Prober` returns `"CP949 (ks_c_5601-1987)"`.

----

It is not related to this story, but I add this just for a reference.
This encoding IS NOT `ks_c_5601-1987` actually, it is superset of `ks_c_5601-1987`. The reason Microsoft used `ks_c_5601-1987` instead of `CP949` is only one: IANA already added `ks_c_5601-1987` to their list. 
(Please refer to [this document](https://lists.w3.org/Archives/Public/ietf-charsets/2002JulSep/0020.html))

----

Thank you for reading this long story and I hope you to understand why I made the decision.

**Comment by rstm-sf** [comment]
@HelloWorld017, thanks for clarification!

The opinion of a native speaker is always good!)

**Comment by rstm-sf** [comment]
@HelloWorld017, I change TODO for cp949. Thanks!

**Comment by rstm-sf** [comment]
@304NotModified, sorry for last change, but it seemed to me that it would be better in future.

File: tests/Charsets.cs [hunk 2]
```diff
@@ -35,22 +35,27 @@
...
-        public const string UTF16_LE = "UTF-16LE";
-        
-        public const string UTF16_BE = "UTF-16BE";
-        
-        public const string UTF32_BE = "UTF-32BE";
-        
-        public const string UTF32_LE = "UTF-32LE";
+        internal const string UTF8 = "utf-8";
 
+        internal const string UTF16_LE = "utf-16le";
```
Comment (304NotModified) [inline tests/Charsets.cs]: Do we still need those?

Comment (304NotModified) [reply tests/Charsets.cs]: We could remove them later, will merge this one first :)

**Comment by 304NotModified** [comment]
thanks!

**Pull request merged**